<a href="https://colab.research.google.com/github/j-claudinei-f/j-claudinei-f/blob/main/Indu%C3%A7%C3%A3oerecorrencia.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#**Definição por recorrência e relação de recorrência**

José Claudinei Ferreira

Universidade Federal de Alfenas

#**Exemplo 1:**

Mostre que, para cada  número natural $n\geq 64$, existem números inteiros positivos $p$ e $q$, tais que $$n=17p+5q.$$

**Demonstração:** Seja $X=\{j\in \mathbb{N}, \text{tais que, para }64\leq k\leq n=(63+j ), \text{ existem números inteiros positivos } p_k ,\, q_k, \text{ com } n=17p_k+5q_k\}$.

Observe que podemos testar a afirmação computacionalmente, para $64\leq n\leq 100$, por exemplo, como segue:

In [6]:
def teste(n):
  sol=[]
  for p in range(0,20):
    for q in range(0,20):
        if (17*p+5*q)==n:
          sol=[p,q]
          break
  return(sol)

Testando

In [7]:
for i in range(63,100+1):
  print('k=',i,', [p,q]=',teste(i))

k= 63 , [p,q]= []
k= 64 , [p,q]= [2, 6]
k= 65 , [p,q]= [0, 13]
k= 66 , [p,q]= [3, 3]
k= 67 , [p,q]= [1, 10]
k= 68 , [p,q]= [4, 0]
k= 69 , [p,q]= [2, 7]
k= 70 , [p,q]= [0, 14]
k= 71 , [p,q]= [3, 4]
k= 72 , [p,q]= [1, 11]
k= 73 , [p,q]= [4, 1]
k= 74 , [p,q]= [2, 8]
k= 75 , [p,q]= [0, 15]
k= 76 , [p,q]= [3, 5]
k= 77 , [p,q]= [1, 12]
k= 78 , [p,q]= [4, 2]
k= 79 , [p,q]= [2, 9]
k= 80 , [p,q]= [0, 16]
k= 81 , [p,q]= [3, 6]
k= 82 , [p,q]= [1, 13]
k= 83 , [p,q]= [4, 3]
k= 84 , [p,q]= [2, 10]
k= 85 , [p,q]= [5, 0]
k= 86 , [p,q]= [3, 7]
k= 87 , [p,q]= [1, 14]
k= 88 , [p,q]= [4, 4]
k= 89 , [p,q]= [2, 11]
k= 90 , [p,q]= [5, 1]
k= 91 , [p,q]= [3, 8]
k= 92 , [p,q]= [1, 15]
k= 93 , [p,q]= [4, 5]
k= 94 , [p,q]= [2, 12]
k= 95 , [p,q]= [5, 2]
k= 96 , [p,q]= [3, 9]
k= 97 , [p,q]= [1, 16]
k= 98 , [p,q]= [4, 6]
k= 99 , [p,q]= [2, 13]
k= 100 , [p,q]= [5, 3]


Como você pode observar acima, é verdade que $j\in X$, para $64\leq n=63+j\leq 100$,  existem números números inteiros positivos $p_n$ e $q_n$, tais que $$n=17p+5q,$$ por checagem.

Vamos supor então que $j\in X$, para algum $n=63+j\geq 80$.

Segue que $64\leq k=n+1-17=n-16\leq n$ pode ser reescrito como $k=17p_k+5q_k$. Então $63+(j+1)n+1=17(p_k+1)+5q_k$, por exemplo;.

Portanto, $j+1\in X$ e $X=\mathbb{N}$, pelo princípio de indução finita. $\square$


**Obs.** Note que a demonstração acima nos dá uma relação de recorrência $$(p_{n+1},q_{n+1})=(p_{n-16}+1,q_{n-16}).$$

Note ainda que há outras formas de determinar $(p_n,q_n)$ aceitos.

#**Exemplo 2:**

Mostre que, para cada  número natural $n\geq 640$, existem números naturais $p$ e $q$, tais que $$n=17p+41q.$$

**Dica para a solução:** Observe que podemos testar a afirmação computacionalmente, para $640\leq n\leq 700$, por exemplo, como segue:

In [14]:
def teste(n):
  sol=[]
  for p in range(0,50):
    for q in range(0,50):
        if (17*p+41*q)==n:
          sol=[p,q]
          break
  return(sol)

Testando

In [17]:
for i in range(639,700+1):
  print('k=',i,', [p,q]=',teste(i))

k= 639 , [p,q]= []
k= 640 , [p,q]= [28, 4]
k= 641 , [p,q]= [16, 9]
k= 642 , [p,q]= [4, 14]
k= 643 , [p,q]= [33, 2]
k= 644 , [p,q]= [21, 7]
k= 645 , [p,q]= [9, 12]
k= 646 , [p,q]= [38, 0]
k= 647 , [p,q]= [26, 5]
k= 648 , [p,q]= [14, 10]
k= 649 , [p,q]= [2, 15]
k= 650 , [p,q]= [31, 3]
k= 651 , [p,q]= [19, 8]
k= 652 , [p,q]= [7, 13]
k= 653 , [p,q]= [36, 1]
k= 654 , [p,q]= [24, 6]
k= 655 , [p,q]= [12, 11]
k= 656 , [p,q]= [0, 16]
k= 657 , [p,q]= [29, 4]
k= 658 , [p,q]= [17, 9]
k= 659 , [p,q]= [5, 14]
k= 660 , [p,q]= [34, 2]
k= 661 , [p,q]= [22, 7]
k= 662 , [p,q]= [10, 12]
k= 663 , [p,q]= [39, 0]
k= 664 , [p,q]= [27, 5]
k= 665 , [p,q]= [15, 10]
k= 666 , [p,q]= [3, 15]
k= 667 , [p,q]= [32, 3]
k= 668 , [p,q]= [20, 8]
k= 669 , [p,q]= [8, 13]
k= 670 , [p,q]= [37, 1]
k= 671 , [p,q]= [25, 6]
k= 672 , [p,q]= [13, 11]
k= 673 , [p,q]= [1, 16]
k= 674 , [p,q]= [30, 4]
k= 675 , [p,q]= [18, 9]
k= 676 , [p,q]= [6, 14]
k= 677 , [p,q]= [35, 2]
k= 678 , [p,q]= [23, 7]
k= 679 , [p,q]= [11, 12]
k= 680 , [p,q]=

Como você pode observar acima, isso é verdade para $640\leq n\leq 700$, por checagem.

Voce pode supor que seja verdade para todo $680\leq k\leq n$. ....

#**Exemplo 3:**

Mostre que, para cada  número natural $n\geq 1000$, existem números naturais $p$ e $q$, tais que $$n=97p+103q.$$

**Dica para a solução:** Observe que podemos testar a afirmação computacionalmente, para $9792\leq n\leq 9900$, por exemplo, como segue:

In [23]:
def teste(n):
  sol=[]
  for p in range(0,300):
    for q in range(0,300):
        if (97*p+103*q)==n:
          sol=[p,q]
          break
  return(sol)

Testando

In [32]:
for i in range(9791,9900+1):
  print('k=',i,', [p,q]=',teste(i))

k= 9791 , [p,q]= []
k= 9792 , [p,q]= [16, 80]
k= 9793 , [p,q]= [33, 64]
k= 9794 , [p,q]= [50, 48]
k= 9795 , [p,q]= [67, 32]
k= 9796 , [p,q]= [84, 16]
k= 9797 , [p,q]= [101, 0]
k= 9798 , [p,q]= [15, 81]
k= 9799 , [p,q]= [32, 65]
k= 9800 , [p,q]= [49, 49]
k= 9801 , [p,q]= [66, 33]
k= 9802 , [p,q]= [83, 17]
k= 9803 , [p,q]= [100, 1]
k= 9804 , [p,q]= [14, 82]
k= 9805 , [p,q]= [31, 66]
k= 9806 , [p,q]= [48, 50]
k= 9807 , [p,q]= [65, 34]
k= 9808 , [p,q]= [82, 18]
k= 9809 , [p,q]= [99, 2]
k= 9810 , [p,q]= [13, 83]
k= 9811 , [p,q]= [30, 67]
k= 9812 , [p,q]= [47, 51]
k= 9813 , [p,q]= [64, 35]
k= 9814 , [p,q]= [81, 19]
k= 9815 , [p,q]= [98, 3]
k= 9816 , [p,q]= [12, 84]
k= 9817 , [p,q]= [29, 68]
k= 9818 , [p,q]= [46, 52]
k= 9819 , [p,q]= [63, 36]
k= 9820 , [p,q]= [80, 20]
k= 9821 , [p,q]= [97, 4]
k= 9822 , [p,q]= [11, 85]
k= 9823 , [p,q]= [28, 69]
k= 9824 , [p,q]= [45, 53]
k= 9825 , [p,q]= [62, 37]
k= 9826 , [p,q]= [79, 21]
k= 9827 , [p,q]= [96, 5]
k= 9828 , [p,q]= [10, 86]
k= 9829 , [p,q]= [27, 

Como você pode observar acima, isso é verdade para $9792\leq n\leq 9900$, por checagem.

Voce pode supor que seja verdade para todo $9990\leq k\leq n$. ....